theseus/src/state/profiles.rs

```diff
@@ -722,7 +736,28 @@ impl Profiles {
                         None
                     }
                 };
-                if let Some(profile) = prof {
+
+                if let Some(mut profile) = prof {
+                    // Clear out modrinth_logs of all files in profiles folder (these are legacy)
+                    // TODO: should be removed in a future build
+                    let modrinth_logs = path.join("modrinth_logs");
+                    if modrinth_logs.exists() {
+                        let _ = std::fs::remove_dir_all(modrinth_logs);
+                    }
+
+                    // For profiles that have locked = None, we assume a legacy profile that hasn't been set yet
+                    if profile.locked.is_none() {
+                        if let Some(ref linked_data) =
+                            profile.metadata.linked_data
+                        {
+                            // If there is a linked project and version, then we assume the profile is locked, as edits have been previously forbidden
+                            profile.locked = Some(
+                                linked_data.project_id.is_some()
+                                    && linked_data.version_id.is_some(),
+                            );
+                        }
+                    }
```

Comment:
this should be able to be removed if you make profile.locked apart of linked data. then you can just make locked true in a serde default so this doesn't have to exist